src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -218,18 +218,21 @@
 
   <!-- archive the test binaries along with some supporting files -->
   <Target Name="ArchiveTestBuild" AfterTargets="CreateAssemblyListTxt" Condition="'$(ArchiveTests)' == 'true'">
+    <PropertyGroup Condition="'$(TestProjectName)'==''">
+      <TestProjectName>$(MSBuildProjectName)</TestProjectName>
+    </PropertyGroup>
     <PropertyGroup>
       <TestArchiveDir>$(TestWorkingDir)$(OSPlatformConfig)/archive/tests/</TestArchiveDir>
       <ProjectJson Condition="!Exists('$(ProjectJson)')">$(OriginalProjectJson)</ProjectJson>
-    </PropertyGroup>
-    <PropertyGroup Condition="'$(TestProjectName)'==''">
-      <TestProjectName>$(MSBuildProjectName)</TestProjectName>
+      <DestinationArchiveFile>$(TestArchiveDir)$(TestProjectName)</DestinationArchiveFile>
+      <DestinationArchiveFile Condition="'$(TargetGroup)' != ''">$(DestinationArchiveFile).$(TargetGroup)</DestinationArchiveFile>
```

Comment:
Does it need to be in the file name? Can it be in the path similar to https://github.com/dotnet/corefx/blob/master/dir.props#L494?